qcodes/instrument/visa.py

```diff
@@ -22,6 +23,14 @@
 log = logging.getLogger(__name__)
 
 
+def _close_visa_handle(
+    handle: pyvisa.resources.MessageBasedResource, name: str
+) -> None:
+    log.info(
+        f"Closing VISA handle to {name} as there are no non weak references to the instrument."
+    )
+    handle.close()
```

Comment:
do we need to implement anything here that would work without issues if the handle has already been closed or even gc-collected? Or is weakref.finalize gracefully taking care of that? I guess, because of this concern the log message goes first so that we are sure to get a message about this event even if handle.close() raises?

Reply:
I don't think so. This has always been called from `__del__` after the instrument was closed but I will check